sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -553,13 +558,43 @@ private <T> Mono<CosmosItemResponse<T>> createItemInternal(T item, CosmosItemReq
                 getId(),
                 database.getId(),
                 database.getClient(),
-                ModelBridgeInternal.getConsistencyLevel(effectiveOptions),
+                effectiveOptions.getConsistencyLevel(),
                 OperationType.Create,
                 ResourceType.Document,
                 requestOptions,
                 trackingId);
     }
 
+    private void applyPolicies(OperationType operationType, ResourceType resourceType, OverridableRequestOptions requestOptions,
+                               String spanName) {
+        CosmosAsyncClient client = this.database.getClient();
+        CosmosDiagnosticsThresholds thresholds = requestOptions != null
+            ? clientAccessor.getEffectiveDiagnosticsThresholds(client, requestOptions.getDiagnosticsThresholds())
+            : clientAccessor.getEffectiveDiagnosticsThresholds(client, null);
+        CosmosDiagnosticsContext cosmosCtx = ctxAccessor.create(
+            spanName,
+            clientAccessor.getAccountTagValue(client),
+            BridgeInternal.getServiceEndpoint(client),
+            database.getId(),
+            getId(),
+            resourceType,
+            operationType,
+            null,
+            clientAccessor.getEffectiveConsistencyLevel(client, operationType, requestOptions.getConsistencyLevel()),
+            null,
+            thresholds,
+            null,
+            clientAccessor.getConnectionMode(client),
+            clientAccessor.getUserAgent(client),
+            null,
+            null,
+            requestOptions);
+
+        CosmosOperationDetails operationDetails = operationDetailsAccessor.create(requestOptions, cosmosCtx);
+        clientAccessor.getOperationPolicies(client).forEach(policy -> policy.process(operationDetails));
```

Comment:
I am wondering should we capture any exceptions when calling `policy.process()`, log it and then translate into a runtime exception

Reply:
Good point

Reply:
Changed to log the exception saying the operation and that is not coming from the sdk